### PR TITLE
drivers/note: unify the spinlock operation in noteram

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
Simplify shared resource protection operations in noteram

Unified use of spinlock_irqsave_wo_note

## Testing
sim:note

### CODE REVIEW

File: drivers/note/note_driver.c [hunk 1]
```diff
@@ -1868,26 +1868,21 @@ int note_get_taskname(pid_t pid, FAR char *buffer)
 {
   FAR struct note_taskname_info_s *ti;
   FAR struct tcb_s *tcb;
-  irqstate_t irq_mask;
 
-  irq_mask = enter_critical_section();
```
Comment (xiaoxiang781216) [inline drivers/note/note_driver.c:1873]: why remove the lock

Comment (Gary-Hobson) [reply drivers/note/note_driver.c]: done

File: drivers/note/noteram_driver.c [hunk 2]
```diff
@@ -289,22 +281,20 @@ static void noteram_remove(void)
...
 
   DEBUGASSERT(buffer != NULL);
-  flags = enter_critical_section();
 
   /* Verify that the circular buffer is not empty */
 
   circlen = noteram_unread_length();
   if (circlen <= 0)
     {
       notelen = 0;
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:296]: return 0;

File: drivers/note/noteram_driver.c [hunk 3]
```diff
@@ -329,7 +319,7 @@ static ssize_t noteram_get(FAR uint8_t *buffer, size_t buflen)
       /* and return an error */
 
       notelen = -EFBIG;
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:321]: return -EFBIG;

File: drivers/note/noteram_driver.c [hunk 4]
```diff
@@ -349,8 +339,7 @@ static ssize_t noteram_get(FAR uint8_t *buffer, size_t buflen)
 
   g_noteram_info.ni_read = read;
 
-errout_with_csection:
-  leave_critical_section(flags);
+errout:
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:342]: remove errout label

File: drivers/note/noteram_driver.c [hunk 5]
```diff
@@ -373,20 +362,17 @@ static ssize_t noteram_get(FAR uint8_t *buffer, size_t buflen)
...
   size_t circlen;
 
-  flags = enter_critical_section();
-
   /* Verify that the circular buffer is not empty */
 
   circlen = noteram_unread_length();
   if (circlen <= 0)
     {
       notelen = 0;
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:374]: return 0;

File: drivers/note/noteram_driver.c [hunk 6]
```diff
@@ -400,8 +386,7 @@ static ssize_t noteram_size(void)
   notelen = note->nc_length;
   DEBUGASSERT(notelen <= circlen);
 
-errout_with_csection:
-  leave_critical_section(flags);
+errout:
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:389]: remove errout

File: drivers/note/noteram_driver.c [hunk 7]
```diff
@@ -427,13 +412,14 @@ static ssize_t noteram_read(FAR struct file *filep,
 {
   ssize_t notelen;
   ssize_t retlen ;
+  int flags;
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:415]: irqstate_t flags

File: drivers/note/noteram_driver.c [hunk 8]
```diff
@@ -485,6 +471,7 @@ static ssize_t noteram_read(FAR struct file *filep,
 static int noteram_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 {
   int ret = -ENOSYS;
+  int flags = spin_lock_irqsave_wo_note(&g_noteram_lock);
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:474]: int->irqstate_t

File: drivers/note/noteram_driver.c [hunk 9]
```diff
@@ -94,9 +94,7 @@ static const struct note_driver_ops_s g_noteram_ops =
   noteram_add
 };
 
-#ifdef CONFIG_SMP
 static volatile spinlock_t g_noteram_lock;
```
Comment (xiaoxiang781216) [inline drivers/note/noteram_driver.c:97]:
```suggestion
static spinlock_t g_noteram_lock;
```

Comment (Gary-Hobson) [reply drivers/note/noteram_driver.c]: done

File: drivers/note/noteram_driver.c (hunk 8 above)
Comment (Gary-Hobson) [reply drivers/note/noteram_driver.c]: done

File: drivers/note/noteram_driver.c (hunk 7 above)
Comment (Gary-Hobson) [reply drivers/note/noteram_driver.c]: done

File: drivers/note/noteram_driver.c (hunk 6 above)
Comment (Gary-Hobson) [reply drivers/note/noteram_driver.c]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**